src/lib.rs

```diff
@@ -432,7 +424,24 @@ where
         // next epoch after it is removed. since it wasn't removed, and the epoch was pinned, that
         // cannot be until after we drop our guard.
         let node = unsafe { node.deref() };
-        let node = node.as_node().unwrap();
+        node.as_node()
```

Comment:
I think this should actually be
```rust
Some(node.as_node().expect("BinEntry::find should always return Node"))
```

Reply:
I had the callers of `get_node` use the `?` operator, but I'm fine with that as well

Reply:
Oh wait, no I kept it as an option because some sanity check in the method might return `None` (see lines 388 and 395).

Reply:
Ah, no, this isn't about the return type of _this_ method (notice that we are re-wrapping in `Some`). It is about having an assertion that checks that `BinEntry::find` _indeed_ always returns a `Node`.

Reply:
(forget all I said)